tests/cloner_test.go

```diff
@@ -531,6 +531,75 @@ var _ = Describe("all clone tests", func() {
 			})
 			ClonerBehavior(cloneStorageClassName, "csivolumeclone")
 		})
+
+		Context("CloneStrategy on storageclass annotation", func() {
+			cloneType := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyCsiClone)
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				storageclass, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if storageclass.GetAnnotations() == nil {
+					storageclass.SetAnnotations(make(map[string]string))
+				}
+
+				storageclass.Annotations["cdi.kubevirt.io/clone-strategy"] = string(cloneType)
+				_, err = f.K8sClient.StorageV1().StorageClasses().Update(context.TODO(), storageclass, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() string {
+					storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					return string(*storageProfile.Status.CloneStrategy)
```

Comment:
Looks like it is possible for storageProfile.Status.CloneStrategy to be nil, I guess if it is nil just return a blank string.
```go
if storageProfile.Status.CloneStrategy == nil {
  return ""
}
return string(*storageProfile.Status.CloneStrategy)
```

Reply:
noted

Reply:
Its the reason the ceph lane failed btw. so it needs to get fixed before it can pass CI

Reply:
right, i have sent in the fix

Reply:
Okay so the test failed again, cloneType is specific type and not string. So it actually found the right value, just in the wrong type, need to cast it to a string.
```
}, time.Minute, time.Second).Should(Equal(string(cloneType)))
```